### PR TITLE
[Transforms] Add more cos combinations to SimplifyLibCalls and InstCombine

### PR DESCRIPTION
**Pull request by AZero13**
Add cos(fabs(x)) -> cos(x) and cos(copysign(x, y)) -> cos(x).

### CODE REVIEW

File: llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp [hunk 1]
```diff
@@ -2488,11 +2488,13 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
...
+    Value *X, *Sign;
     Value *Src = II->getArgOperand(0);
-    if (match(Src, m_FNeg(m_Value(X))) || match(Src, m_FAbs(m_Value(X)))) {
-      // cos(-x) -> cos(x)
-      // cos(fabs(x)) -> cos(x)
+    if (match(Src, m_FNeg(m_Value(X))) || match(Src, m_FAbs(m_Value(X))) ||
+        match(Src, m_CopySign(m_Value(X), m_Value(Sign)))) {
+      // cos(-x) --> cos(x)
+      // cos(fabs(x)) --> cos(x)
+      // cos(copysign(x, y)) --> cos(x)
```
Comment (jcranmer-intel) [inline llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp:2497]: I think this could be broken out into an `optimizeEvenFunction` helper method.

Comment (AtariDreams) [reply llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp:2497]: I'll do that in another PR. I think this is good for now.

File: llvm/test/Transforms/InstCombine/cos-1.ll [hunk 2]
```diff
@@ -100,6 +108,71 @@ define float @cosf_unary_negated_arg_FMF(float %x) {
...
+define float @cosf_copysign_arg_FMF(float %x, float %y) {
+; ANY-LABEL: @cosf_copysign_arg_FMF(
+; ANY-NEXT:    [[COS:%.*]] = call reassoc nnan float @cosf(float [[X:%.*]])
+; ANY-NEXT:    ret float [[COS]]
+;
+  %copysign = tail call float @llvm.copysign.f32(float %x, float %y)
+  %r = call nnan reassoc float @cosf(float %copysign)
+  ret float %r
+}
+
```
Comment (arsenm) [inline llvm/test/Transforms/InstCombine/cos-1.ll:175]: test a vector intrinsic case?

File: llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp (hunk 1 above)
Comment (arsenm) [reply llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp:2497]: In InstCombineAndOrXor we have stripSignOnlyFPOps

**Review by arsenm — APPROVED** [review]
LGTM, but could try to common with stripSignOnlyFPOps in a follow up

**Pull request merged**